chia/wallet/nft_wallet/nft_puzzles.py

```diff
@@ -207,6 +209,53 @@ def get_percentage_from_puzzle(puzzle: Program) -> Optional[uint64]:
     return None
 
 
+def get_nft_info_from_puzzle(puzzle: Program, nft_coin: Coin) -> NFTInfo:
+    """
+    Extract NFT info from a full puzzle
+    :param puzzle: NFT full puzzle
+    :param nft_coin: NFT coin
+    :return: NFTInfo
+    """
+    # TODO Update this method after the NFT code finalized
+    curried_args = match_nft_puzzle(puzzle)[1]
```

Comment:
Should probably check that the boolean returned by match_nft_puzzle(puzzle)[0] is actually True before attempting to pull values out of the returned iterator. 

```suggestion
    match = match_nft_puzzle(puzzle)
    if not match[0]:
        raise ValueError("Unable to match NFT puzzle")
    curried_args = match_nft_puzzle(puzzle)[1]
```

Reply:
This is why returning tuples and returning statuses are bad.  The code outside is illegible (what is element 1?) and errors are easily not propagated (the status ceases to be available after indexing the result).  These are general issues we have so not really picking out this code in particular beyond being the example at hand now.

Also, instead of adding a `raise` here, and at every other call, we could just raise inside `match_nft_puzzle()`.  And when you do need to return multiple values, write a dataclass so the values are accessible by name and clearly defined.

Reply:
I agree with you. I saw this pattern in CAT wallet and many other places so I thought it's a common thing in python.

Reply:
For the raise I think not all the use cases want to handle the exception. In some cases they just want a simple if check and move on instead of failing the whole function.

Reply:
Well if the match fails, an exception will be raised regardless with the current code. My point was to at least make it a bit more obvious what the failure was (that the match failed) rather than getting an exception like `ValueError: not enough values to unpack (expected 6, got 0)`

Reply:
I agree with @altendky that it's probably better to fix `match_nft_puzzle`.

Reply:
They can catch the exception and handle it.  But sure, there's some balance in deciding the meaning of the return of the function.  It may be that it should just return a list.  If nothing is found the list is empty.  I would have to learn more about the `match_nft_puzzle()` function to weigh in in more detail about that.

I don't think this is particularly common in Python, but it certainly is common here.  It's one of the points I think we need to discuss and come to a conclusion about our approach.